raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -103,6 +111,58 @@ public void testInitializeSingleMemberQuorum() throws IOException {
         assertEquals(0, channel.drainSendQueue().size());
     }
 
+    @Test
+    public void testInitializeAsLeaderFromStateStore() throws IOException {
+        Set<Integer> voters = Utils.mkSet(localId, 1);
+        MockQuorumStateStore electionStore = new MockQuorumStateStore() {
+            @Override
+            public ElectionState readElectionState() {
+                return ElectionState.withElectedLeader(2, localId);
+            }
+        };
+        KafkaRaftClient client = buildClient(voters, electionStore);
+        assertTrue(stateMachine.state().isPresent());
+        assertTrue(stateMachine.state().get() instanceof LeaderState);
+        assertTrue(client.quorumState().isLeader());
+        assertEquals(1L, log.endOffset());
+
+        // Should have sent out connection info query for other node id.
```

Comment:
Some of this is covered in `assertSentFindQuorumRequest`. Any reason not to use/extend that?

Reply:
Good call, will take a look.